### ISSUE TITLE
[assemble] JLink assembler launcher missing main jar

### ISSUE THREAD

**Issue by jvalkeal** [jreleaser/jreleaser#1496]
### Task List

- [ ] Steps to reproduce provided
- [ ] Stacktrace (if present) provided
- [ ] Example that reproduces the problem (link to git repository is ideal)
- [x] Full description of the issue provided (see below)

### Steps to Reproduce

When jlink is assembled a template variable `distributionJavaMainJar` in https://github.com/jreleaser/jreleaser/blob/e4a487d85d5901bf2b46b45cc46cbcf8f512dd33/core/jreleaser-templates/src/main/resources/META-INF/jreleaser/templates/jlink/jlink/bin/launcher.tpl#L142 is empty, thus it creates a non-working launcher script. Same type of launcher script in `JavaArchive` works fine what comes for main jar.

I was looking differences between `JlinkAssemblerProcessor` and `JavaArchiveAssemblerProcessor` and latter seems to set properties in https://github.com/jreleaser/jreleaser/blob/e4a487d85d5901bf2b46b45cc46cbcf8f512dd33/core/jreleaser-engine/src/main/java/org/jreleaser/assemblers/JavaArchiveAssemblerProcessor.java#L97 while former don't do no customisation for template properties.

I compiled 1.9.0 snapshot locally and modified `JlinkAssemblerProcessor` to set those properties similarly to `JavaArchiveAssemblerProcessor` and it did fix the issue. 

### Expected Behaviour

JLink should have working launcher script with fatjar.

Something like:
```
set -- \
        -classpath "$CLASSPATH" \
        -jar "${JARSDIR}/spring-cli-0.7.1.jar" \
        "$@"
```

### Actual Behaviour

JLink launcher is missing path to an actual jar file.

```
set -- \
        -classpath "$CLASSPATH" \
        -jar "${JARSDIR}/" \
        "$@"
```

This is for spring-cli where I'm trying to add "standalone" release similarly to JReleaser itself. Config file is https://github.com/spring-projects-experimental/spring-cli/blob/main/jreleaser.yml

**Comment by jvalkeal** [comment]
@aalmiray I could try to PR this fix.

**Comment by aalmiray** [comment]
Thank you @jvalkeal. Be aware that perhaps several templates may need an update as well.

**Comment by aalmiray** [comment]
🎉 This issue has been resolved in `v1.9.0` ([Release Notes](https://github.com/jreleaser/jreleaser/releases/tag/v1.9.0))
